src/main/java/pathfinding/BreadthFirstGraph.java

```diff
@@ -0,0 +1,49 @@
+package pathfinding;
+
+import java.util.*;
+
+public class BreadthFirstGraph implements Graph {
+
+    public ArrayList<Node> findPath(Node start, Node end) {
+
+        ArrayList<Node> result = new ArrayList<Node>();
+        HashMap<Node, Node> parentMap = new HashMap<Node, Node>();
+        parentMap.put(start, null);
+        LinkedList<Node> nodeQueue = new LinkedList<Node>();
+
+        if (start == null || end == null)
+            return null;
+
+        if (start == end) {
+            ArrayList<Node> ret = new ArrayList<>();
+            ret.add(start);
+            return ret;
+        }
+
```

Comment:
DEFINITELY needs more comments (there's not a single comment in this file...) but I care more about getting this merged ASAP than I do about comments.... lucky for you.